### PR TITLE
Fix datasets.models.covariance after fit to match FitResults.models.covariance

### PR DESCRIPTION
**Pull request by QRemy**
Fix datasets.models.covariance after fit to match FitResults.models.covariance.
It simply add a  private `_covariance` attribute on datasets that will be used to set the covariance properly in `.models` property  if the parameters of the covariance and the datasets.models match.

### CODE REVIEW

**Review by bkhelifi — DISMISSED** [review]
Thanks a lot. It looks good! Apparently an if statement is not tested correctly. Could you add a test, just to be sure? The issue is an important one and one should be sure to fix it definitely

**Comment by QRemy** [comment]
>  Could you add a test, just to be sure? The issue is an important one and one should be sure to fix it definitely

Done also modified a bit the code so the solution for actors is simpler.

File: gammapy/datasets/core.py [hunk 1]
```diff
@@ -178,6 +183,8 @@ def models(self, models):
         Duplicate model objects have been removed.
```
Comment (bkhelifi) [inline gammapy/datasets/core.py:183]: For me, the docstring is a bit obscure... The setter is about unique models: what is a unique model? The docstring describes in fact what does the setter and the input parameter is not precised. Can you enlight a bit this docstring?

**Review by bkhelifi — APPROVED** [review]
In between the different PRs about covariance, it is a bit hard to have a global view. However, it seems wise and looks ok for me.
I have only a minor comment on a docstring.

File: gammapy/datasets/core.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/core.py:183]: The setter just display the same output as its associated property. I think we have the same behaviour everywhere. Probably better not change too much docstring in a bugfix that has to beck backported.

**Pull request merged**